### PR TITLE
test(integration): add widget-level integration test suite for user app navigation

### PR DESCRIPTION
**Pull request by Mark-Yun**
## Summary
- `Supabase.instance` 직접 접근 3개 파일에서 Riverpod provider 경유로 리팩토링 (동작 변경 없음)
- Mocked integration test harness 구축 (`test_app.dart`, `test_mocks.dart`, `mock_data.dart`)
- 핵심 CUJ integration test 12개 파일 추가:
  - 로그인/redirect 플로우 (`auth_redirect_test`, `login_page_test`)
  - 홈/탐색 (`home_navigation_test`, `search_flow_test`)
  - 이벤트 상세 (`cuj_event_detail_test`)
  - 이벤트 신청 (`cuj_event_application_test`)
  - 마이페이지 (`my_page_test`)
  - 알림 (`cuj_notification_test`)
  - Edge cases (`edge_cases_test`)
- CI pipeline에 integration test step 추가 (`app_user` only)

## Refactoring (3 files, ~6 lines)
| 파일 | 변경 |
|------|------|
| `app_router.dart` | `Supabase.instance.client.auth.currentUser` → `ref.read(currentUserProvider)` |
| `user_profile_provider.dart` | 직접 쿼리 → `userRepositoryProvider.getUserProfile()` |
| `social_interaction_controller.dart` | `Supabase.instance.client.auth.currentUser` → `ref.read(currentUserProvider)` |

## Test Strategy
- Riverpod `ProviderScope(overrides: [...])` 로 모든 외부 의존성 mock
- `pumpAndSettle()` 미사용 — `pump()` only (무한 애니메이션 timeout 방지)
- `Supabase.initialize()` / `Firebase.initializeApp()` 호출 없이 headless 실행
- 한국어 fixture data (김민글, 강남 소셜 파티 등)

### CODE REVIEW

File: apps/app_user/integration_test/cuj_notification_test.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  group('Notification Center CUJ', () {
+    testWidgets('알림 목록 렌더링', (tester) async {
+      setKoreanLocale(tester);
+      final fakeUser = createFakeUser();
+      final mockNotificationRepo = MockNotificationRepository();
+      when(
+        mockNotificationRepo.getNotifications,
+      ).thenAnswer((_) async => createFakeNotifications());
```
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/integration_test/cuj_notification_test.dart]: This file (`cuj_notification_test.dart`) was removed in the force-pushed cleanup. It no longer exists in the current PR. Not applicable.

[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/integration_test/cuj_notification_test.dart]: This file (cuj_notification_test.dart) was removed in the force-pushed cleanup. It no longer exists in the current PR. Not applicable.

File: apps/app_user/lib/src/routing/app_router.dart [hunk 2]
```diff
@@ -25,7 +25,7 @@ GoRouter goRouter(Ref ref) {
     refreshListenable: authState,
     redirect: (context, state) {
       // Supabase SDK에서 직접 읽음 — Riverpod 캐시 갱신 타이밍과 무관하게 정확
-      final isLoggedIn = Supabase.instance.client.auth.currentUser != null;
+      final isLoggedIn = ref.read(currentUserProvider) != null;
```
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/lib/src/routing/app_router.dart]: This file (app_router.dart) was not modified in the current PR. The comment refers to a previous (reverted) commit that was cleaned up in the force push. Not applicable.

File: apps/app_user/integration_test/utils/test_app.dart [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+import 'package:minglit_kit/minglit_kit.dart';
+
+/// Integration test용 TestApp 헬퍼.
+/// 실제 Supabase/Firebase 없이 GoRouter 네비게이션을 테스트합니다.
+Widget createTestApp({
+  bool isLoggedIn = false,
+  User? currentUser,
```
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/integration_test/utils/test_app.dart:15]: Intentional design: `isLoggedIn` controls GoRouter redirect logic (route-level auth gate), while `currentUser` controls what the UI renders (avatar, notifications icon). In practice, tests always pass both consistently (e.g., `isLoggedIn: true, currentUser: user`), but keeping them separate allows testing edge cases like 'what does the UI show if the user object is null but redirects think we are logged in'. That said, the suggestion to unify via `effectiveLoggedIn` is valid for safety — will consider applying in a follow-up if this separation causes confusion.
